docs/orchestrating-elastic-stack-applications/security/users-and-roles.asciidoc

```diff
@@ -55,7 +56,30 @@ spec:
 You can reference several secrets in the Elasticsearch specification.
 ECK aggregates their content into a single secret, mounted in every Elasticsearch Pod.
 
-Referenced secrets may be composed of 2 entries. You can provide either entry or both entry in each secret:
+Referenced secrets may be of one of two types:
+
+
+. a combination of username and password as in link:https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret[Kubernetes basic authentication secrets]
+. a raw file realm content secret
+
+A basic authentication secret can optionally also contain a `users_roles` file. To associate one or more roles with the user, as described in the link:https://www.elastic.co/guide/en/elasticsearch/reference/current/file-realm.html[file realm documentation]. The following example illustrates this combination:
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-basic-auth
+type: kubernetes.io/basic-auth
+stringData:
+  username: rdeniro    # required field for kubernetes.io/basic-auth
+  password: mypassword # required field for kubernetes.io/basic-auth
+  users_roles: |-
+    admin:rdeniro  # optional, not part of kubernetes.io/basic-auth
+----
+
+NOTE: If you specify the password for the `elastic` user through such a basic authentication secret then the secret holding the password described in <<{p}-default-elastic-user>> will not be created by the operator.
+
```

Comment:
Maybe add a note that it can take some time (~1min) befire it's available?

Reply:
Is that not true for everything we orchestrate with ECK? Everything is asynchronous and available after a variable delay.